lib/connections/connections.go

```diff
@@ -35,17 +37,39 @@ var (
 	listeners = make(map[string]ListenerFactory, 0)
 )
 
+type ConnectionHandlingModel interface {
```

Comment:
I suggest calling this just `Model`, or maybe `ConnectionHandler` (that's closer to what it does form the point of view of this package), and it should just include the methods this package needs.